apps/src/code-studio/components/progress/ResourcesDropdown.jsx

```diff
@@ -32,36 +32,60 @@ export default class ResourcesDropdown extends React.Component {
   handleDropdownClick = () => {
     if (this.props.unitGroupId) {
       this.recordFirehose('unit-group', 'click-dropdown', {
-        unitGroupId: this.props.unitGroupId
+        unitGroupId: this.props.unitGroupId,
+        studentFacing: !!this.props.studentFacing
       });
     } else if (this.props.unitId) {
       this.recordFirehose('unit', 'click-dropdown', {
-        unitId: this.props.unitId
+        unitId: this.props.unitId,
+        studentFacing: !!this.props.studentFacing
       });
     }
   };
 
-  handleItemClick = () => {
+  handleItemClick = (e, resource) => {
+    // Needed so that we can keep the href on the link to allow for standard link interactions
+    e.preventDefault();
+    const callback = () => {
+      window.open(resource.url, 'noopener', 'noreferrer');
+    };
+    const resourceKey = this.props.useMigratedResources
+      ? resource.key
+      : resource.type;
     if (this.props.unitGroupId) {
-      this.recordFirehose('unit-group', 'click-resource', {
-        unitGroupId: this.props.unitGroupId
-      });
+      this.recordFirehose(
+        'unit-group',
+        'click-resource',
+        {
+          unitGroupId: this.props.unitGroupId,
+          resourceKey: resourceKey,
+          studentFacing: !!this.props.studentFacing
+        },
+        callback
+      );
     } else if (this.props.unitId) {
-      this.recordFirehose('unit', 'click-resource', {
-        unitId: this.props.unitId
-      });
+      this.recordFirehose(
+        'unit',
+        'click-resource',
+        {
+          unitId: this.props.unitId,
+          resourceKey: resourceKey,
+          studentFacing: !!this.props.studentFacing
+        },
+        callback
+      );
     }
   };
 
-  recordFirehose = (study_group, event, data_json) => {
+  recordFirehose = (study_group, event, data_json, callback) => {
     firehoseClient.putRecord(
       {
         study: 'teacher-resources',
```

Comment:
is it worth having a different study for student resources?

Reply:
I liked the idea of having them in the same study in case the same resource was both a student and teacher resource but `teacher-resources` is misleading enough that I think having a separate study would be good for now.